src/payment/strategies/bolt/bolt-app-payment-strategy.spec.ts

```diff
@@ -0,0 +1,187 @@
+      paymentHumanVerificationHandler = new PaymentHumanVerificationHandler(createSpamProtection(createScriptLoader()));
+      paymentActionCreator = new PaymentActionCreator(
+          paymentRequestSender,
+          orderActionCreator,
+          paymentRequestTransformer,
+          paymentHumanVerificationHandler
+      );
+      submitOrderAction = of(createAction(OrderActionType.SubmitOrderRequested));
+      submitPaymentAction = of(createAction(PaymentActionType.SubmitPaymentRequested));
+      paymentMethodRequestSender = new PaymentMethodRequestSender(requestSender);
+      paymentMethodActionCreator = new PaymentMethodActionCreator(paymentMethodRequestSender);
+      boltClient = getBoltScriptMock(true);
+      boltScriptLoader = new BoltScriptLoader(scriptLoader);
+
+      payload = {
+          payment: {
+              methodId: 'bolt',
+              paymentData: {
+                  nonce: 'fooNonce',
+              },
+          },
+      };
+      options = { methodId: 'bolt' };
+
+      jest.spyOn(store, 'dispatch');
+
+      jest.spyOn(orderActionCreator, 'submitOrder')
+          .mockReturnValue(submitOrderAction);
+      jest.spyOn(paymentActionCreator, 'submitPayment')
+          .mockReturnValue(submitPaymentAction);
+      jest.spyOn(paymentMethodActionCreator, 'loadPaymentMethod')
+          .mockResolvedValue(store.getState());
+      jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+          .mockReturnValue(paymentMethodMock);
+      jest.spyOn(boltScriptLoader, 'load')
+          .mockResolvedValue(boltClient);
+
+      strategy = new BoltAppPaymentStrategy(
+        store,
+        orderActionCreator,
+        paymentActionCreator,
+        paymentMethodActionCreator,
+        boltScriptLoader
+      );
+  });
+
+  describe('#initialize', () => {
+    it('successfully initializes the bolt strategy and loads the bolt client', async () => {
+        await expect(strategy.initialize(options)).resolves.toEqual(store.getState());
+        expect(boltScriptLoader.load).toHaveBeenCalledWith('publishableKey', true);
+    });
+
+    it('fails to initialize the bolt strategy if no publishableKey is not provided', async () => {
+      paymentMethodMock.initializationData.publishableKey = null;
+      await expect(strategy.initialize(options)).rejects.toThrow(MissingDataError);
+      expect(boltScriptLoader.load).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('#execute', () => {
+    let options: PaymentInitializeOptions;
+    const expectedPayment = {
+      methodId: 'bolt',
+      paymentData: {
+          nonce: 'transactionReference',
+      },
+    };
+    options = {
+      methodId: 'bolt',
+    };
+
+    it('successfully executes the bolt strategy and submits payment', async () => {
+        await strategy.initialize(options);
+        await strategy.execute(payload);
+        expect(boltClient.configure).toHaveBeenCalledWith(expect.any(Object), {}, expect.any(Object));
```

Comment:
instead of `expect.any(Object)` could you be more explicit and use something like

`expect.objectContaining({ .... })`